packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,303 @@
+		setShowCommentBoard( true );
+		openGeneralSidebar( 'edit-post/collab-sidebar' );
+	};
+
+	// Function to save the comment.
+	const addNewComment = async ( comment, parentCommentId ) => {
+		const args = {
+			post: postId,
+			content: comment,
+			comment_type: 'block_comment',
+			comment_approved: 0,
+		};
+
+		if ( parentCommentId ) {
+			args.parent = parentCommentId;
+		}
+
+		const savedRecord = await saveEntityRecord( 'root', 'comment', args );
+
+		if ( savedRecord ) {
+			// If it's a main comment, update the block attributes with the comment id.
+			if ( ! parentCommentId ) {
+				updateBlockAttributes( clientId, {
+					blockCommentId: savedRecord?.id,
+				} );
+			}
+
+			createNotice(
+				'snackbar',
+				parentCommentId
+					? // translators: Reply added successfully
+					  __( 'Reply added successfully.' )
+					: // translators: Comment added successfully
+					  __( 'Comment added successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+			fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onCommentResolve = async ( commentId ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			status: 'approved',
+		} );
+
+		if ( savedRecord ) {
+			// translators: Comment resolved successfully
+			createNotice( 'snackbar', __( 'Thread marked as resolved.' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+
+			fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onEditComment = async ( commentId, comment ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			content: comment,
+		} );
+
+		if ( savedRecord ) {
+			createNotice(
+				'snackbar',
+				// translators: Comment edited successfully
+				__( 'Thread edited successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+
+			fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onError = () => {
+		createNotice(
+			'error',
+			// translators: Error message when comment submission fails
+			__(
+				'Something went wrong. Please try publishing the post, or you may have already submitted your comment earlier.'
+			),
+			{
+				isDismissible: true,
+			}
+		);
+	};
+
+	const onCommentDelete = async ( commentId ) => {
+		await deleteEntityRecord( 'root', 'comment', commentId );
+
+		updateBlockAttributes( clientId, {
+			blockCommentId: undefined,
+		} );
+
+		createNotice(
+			'snackbar',
+			// translators: Comment deleted successfully
+			__( 'Thread deleted successfully.' ),
+			{
+				type: 'snackbar',
+				isDismissible: true,
+			}
+		);
+		fetchComments();
+	};
+
+	const fetchComments = useCallback( async () => {
+		if ( postId ) {
+			const data = await getEntityRecords( 'root', 'comment', {
+				post: postId,
+				type: 'block_comment',
+				status: 'any',
+				per_page: 100,
+			} );
+
+			if ( data ) {
+				setThreads( Array.isArray( data ) ? data : [] );
```

Comment:
Why is the threads state not derived from the entities (using the entities selector)?